src/gwt/src/org/rstudio/studio/client/workbench/views/source/Source.java

```diff
@@ -723,15 +726,16 @@ else if (uiPrefs_.enableEmacsKeybindings().getGlobalValue())
       
       // handle chunk options event
       handleChunkOptionsEvent();
+   }
+   
+   private boolean consoleEditorHadFocusLast()
+   {
+      AceEditor editor = editorFocusTracker_.getLastFocusedEditor();
```

Comment:
Does this need to work across source windows? (If so, look at `DocFocusedEvent` and `SourceWindowManager.getLastFocusedSourceWindowId`)